src/models/appmodel.ts

```diff
@@ -13,7 +13,23 @@ export class AppModel implements IAppModel {
     get selectedFlavorIndex(): number { return this._selectedFlavorIndex; }
     set selectedFlavorIndex(value: number) { this._selectedFlavorIndex = value; }
 
+    // we can create a dedicated Class for SelectedItem like: class SelectedItem etc. but it will be too complicated
+    @observable private _selectedItem = { label: "", value: "" };
+    get selectedItem(): ComboboxItem { return this._selectedItem; }
+    set selectedItem(item: ComboboxItem) { this._selectedItem = item; }
+
     @observable private _textInput = "";
     get textInput(): string { return this._textInput; }
     set textInput(value: string) { this._textInput = value; }
+    get textInputLength(): number { return this._textInput.length; }
+
+    @observable private _transformationText = "Here is some dummy text.";
```

Comment:
The buttons should act on _textInput.   Remove _transformationText

Reply:
Done
Attacked the demo gif to the Trello card